src/xdealer_channel.cpp

```diff
@@ -0,0 +1,57 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xdealer_channel.hpp"
+
+namespace xeus
+{
+
+    xdealer_channel::xdealer_channel(zmq::context_t& context,
+                                 const xeus::xconfiguration& config)
+        : m_socket(context, zmq::socket_type::dealer)
+        , m_end_point("")
+    {
+        // TODO
+        m_end_point = xeus::get_end_point(config.m_transport, config.m_ip, config.m_shell_port);
+
+        m_socket.connect(m_end_point);
+    }
+
+    xdealer_channel::~xdealer_channel()
+    {
+        m_socket.disconnect(m_end_point);
+    }
+
+    void xdealer_channel::send_message(zmq::multipart_t& message)
+    {
+        send_message_impl(message, m_socket);
+    }
+
+    std::optional<zmq::multipart_t> xdealer_channel::receive_message(long timeout)
+    {
+        return receive_message_impl(timeout, m_socket);
```

Comment:
Same here, no need to a receive_message_impl, you can implement the method directly.